presto-main/src/main/java/com/facebook/presto/execution/StageStateMachine.java

```diff
@@ -326,6 +334,11 @@ public BasicStageStats getBasicStageStats(Supplier<Iterable<TaskInfo>> taskInfos
 
     public StageInfo getStageInfo(Supplier<Iterable<TaskInfo>> taskInfosSupplier)
```

Comment:
Curious: Why it takes `Supplier` as the input?

Reply:
That is a very good question.  The "supplierness" of this and `getBasicStageStats` doesn't seem to be used at all.  I'll address that in another PR.